pkg/util/predicates.go

```diff
@@ -0,0 +1,101 @@
+package util
+
+import (
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ComposePredicates will compose a variable number of predicte and return a predicate that
+// will allow events that are allowed by any of the given predicates.
+func ComposePredicates(predicates ...predicate.Predicate) predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			for _, p := range predicates {
+				if p != nil && p.Create(e) {
+					return true
+				}
+			}
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			for _, p := range predicates {
+				if p != nil && p.Delete(e) {
+					return true
+				}
+			}
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			for _, p := range predicates {
+				if p != nil && p.Update(e) {
+					return true
+				}
+			}
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			for _, p := range predicates {
+				if p != nil && p.Generic(e) {
+					return true
+				}
+			}
+			return false
+		},
+	}
+}
+
+// LabelsChangedPredicate will only allow events that changed Metadata.Labels
+type LabelsChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements the update event trap for LabelsChangedPredicate
+func (p LabelsChangedPredicate) Update(e event.UpdateEvent) bool {
+	return e.MetaOld != nil &&
+		e.MetaNew != nil &&
+		!reflect.DeepEqual(e.MetaOld.GetLabels(), e.MetaNew.GetLabels())
+}
+
+// FinalizersChangedPredicate will only allow events that changed Metadata.Finalizers
+type FinalizersChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements the update event trap for LabelsChangedPredicate
+func (p FinalizersChangedPredicate) Update(e event.UpdateEvent) bool {
```

Comment:
change the comment to // Update implements the update event trap for FinalizersChangedPredicate

Reply:
done